### DISCUSSION TITLE
Project Deploy PHP on Vercel

### DISCUSSION THREAD

**Discussion by leandroslopes** [vercel/community#832] [Help]
I am deploying a Pure PHP project and get this message:

> Error: No more than 12 serverless functions can be added to a deployment on the Hobby plan. Create a team (Pro plan) to deploy more. Learn more: https://vercel.link/function-count-limit

Does anyone have any idea how to get around this situation? Or can you only solve this error by purchasing a paid plan from Vercel?

[![memcel-project.png](https://i.postimg.cc/bY4SbqXf/memcel-project.png)](https://postimg.cc/Wt64cLb5)

Project repository: https://github.com/leandroslopes/memcel

**Comment by amyegan** [comment]
Hi @leandroslopes. 12 Serverless Functions Created per Deployment is the [limit](https://vercel.com/docs/concepts/limits/overview#general-limits) for Hobby accounts, but Pro would remove that limit. 

I'm not very familiar with PHP development, so I may be misunderstanding, but I suspect that the contents of your `/api` directory aren't all intended to be Serverless Functions. If so, changing up your project's structure a bit might help (source: https://github.com/vercel/vercel/discussions/4983)

There is theoretically a way around the 12 Serverless Functions limit by bundling since [only dedicated Lambdas are counted as one Serverless Function](https://vercel.com/docs/concepts/limits/overview#serverless-functions-created-per-deployment). With Next.js, Serverless Functions are combined as much as possible into fewer Lambdas. Automatically [bundling Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions#bundling-serverless-functions) is currently only enabled for Next.js, but will be enabled for other scenarios in the future.

**Reply by leandroslopes** [reply]
Hey @amyegan, thanks for the help!  But in the link you gave me the example is from a JavaScript application.  My application on Vercel is PHP, and it is exactly as per the repository link and image in the discussion.  Let's see if anyone else can help me with something more precise.

**Reply by amyegan** [reply]
You may have already seen it, but the GitHub repo for the [PHP Runtime](https://github.com/vercel-community/php) has linked some [examples](https://github.com/vercel-community/php#examples) and [resources](https://github.com/vercel-community/php#-resources) that could be helpful. 

Are you intending to have more than 12 serverless functions? Or is it only supposed to be the one you specified in your `vercel.json`? It looks like some files could be unintentionally increasing the number because contents of the `/api` directory are automatically deployed as serverless functions:
>By default, no configuration is needed to deploy Serverless Functions to Vercel.
>
>For all officially supported runtimes, the only requirement is to create an api directory at the root of your project directory, placing your Serverless Functions inside.
>
> – [Project config documentation](https://vercel.com/docs/project-configuration#project-configuration/functions/description)

Hopefully someone more familiar with the PHP Runtime can provide a more guidance here.

**Reply by leandroslopes** [reply]
@amyegan what in fact are serverless functions? 

Is every .php file treated as a serverless function?

I have placed my entire Pure PHP project in the /api folder as per the example in the GitHub repo.

I am confused!

**Reply by amyegan** [reply]
Sorry for the delayed response! I could not find an example repo with the entire app living in the `/api` directory. The ones I'm seeing have only an `index.php` in that directory and locate all of the other files in other directories. Can you share the example you are following?

It might help to get a general understanding of Serverless Functions and how they work on Vercel. The Vercel docs use Next.js and Node.js for example code, but the concepts carry over to any supported language.
- [Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions#)
- [Customizing Serverless Functions](https://vercel.com/blog/customizing-serverless-functions)

**Comment by Zeit08** [comment]
Have you solved it? I have a similar issue where I have my static site with a contact form. As far as I got is the same example you mentioned, so I to put my contact-form.php in the /api folder, and use the community php runtime. But I am a newbie in this and I have not found a precise step-by-step instructions of how to use that php file on my static vercel site.

**Reply by amyegan** [reply]
Hi @Zeit08! I don't know a whole lot about the PHP runtime, but the repo has a nice [readme](https://github.com/vercel-community/php#readme) with a lot of helpful information. The [FAQ section](https://github.com/vercel-community/php#%EF%B8%8F-faq) gives some more detail about how to use more than one endpoint and other common questions. 

There are a few good [example apps](https://github.com/vercel-community/php#examples) linked in the readme that you can use as a model for your own app. You'll see that all of them have only a single index.php file in `/api`. The rest of the files live in other directories so they are not mistakenly deployed as serverless functions.
